### ISSUE TITLE
[Bug] [RLlib] CQL does not support complex observations

### ISSUE THREAD

**Issue by fwitter** [ray-project/ray#23331]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### Issue Severity

Medium: It contributes to significant difficulty to complete my task but I work arounds and get it resolved.

### What happened + What you expected to happen

When training CQL with complex observations, i.e. structures consisting of `gym.spaces.Dict` or `gym.spaces.Tuple`, `CQLTrainer` throws an error during loss computation

### Versions / Dependencies

Ray Nightly

### Reproduction script

```python
from ray.rllib.agents import cql


# Custom CQLTainer config
config = {}
config['observation_space'] = gym.spaces.Dict({
    'SV': gym.spaces.Box(shape=(1,)),
    'CV': gym.spaces.Box(shape=(3, 200, 200))
})
config['action_space'] = gym.spaces.Box(shape=(2,))
config['Q_model'] = {
    'custom_model': 'ComplexInputNetwork',  # Taken from docs at https://docs.ray.io/en/latest/rllib/rllib-models.html#more-examples-for-building-custom-models
    'conv_filters': [[4, 5, 1], [8, 10, 1]],
    'fcnet_hiddens': [256, 256, 256],
}
config['policy_model'] = {
    'custom_model': 'ComplexInputNetwork',
    'conv_filters': [[4, 5, 1], [8, 10, 1]],
    'fcnet_hiddens': [256, 256, 256],
}
config['input'] = '/path/to/my_offline_rl_data.json'


trainer = cql.CQLTrainer(config=config)
for i in range(10):
    trainer.train()
```

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!
